src/gt4py/next/program_processors/runners/dace_fieldview/utility.py

```diff
@@ -89,3 +92,43 @@ def get_map_variable(dim: gtx_common.Dimension) -> str:
     """
     suffix = "dim" if dim.kind == gtx_common.DimensionKind.LOCAL else ""
     return f"i_{dim.value}_gtx_{dim.kind}{suffix}"
+
+
+def get_tuple_fields(
+    tuple_name: str, tuple_type: ts.TupleType, flatten: bool = False
+) -> list[tuple[str, ts.DataType]]:
+    """
+    Creates a list of names with the corresponding data type for all elements of the given tuple.
+
+    Examples
+    --------
+    >>> sty = ts.ScalarType(kind=ts.ScalarKind.INT32)
+    >>> fty = ts.FieldType(dims=[], dtype=ts.ScalarType(kind=ts.ScalarKind.FLOAT32))
+    >>> t = ts.TupleType(types=[sty, ts.TupleType(types=[fty, sty])])
+    >>> assert get_tuple_fields("a", t) == [("a_0", sty), ("a_1", ts.TupleType(types=[fty, sty]))]
+    >>> assert get_tuple_fields("a", t, flatten=True) == [
+    ...     ("a_0", sty),
+    ...     ("a_1_0", fty),
+    ...     ("a_1_1", sty),
+    ... ]
+    """
+    fields = [(f"{tuple_name}_{i}", field_type) for i, field_type in enumerate(tuple_type.types)]
```

Comment:
Can this collide with a tuple called `{tuple_name}_0`? Maybe prefix with `_<something>_`

Reply:
Name collision will be detected when we try to add scalar/array storage to the SDFG with this name.